plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java

```diff
@@ -50,14 +51,36 @@ protected String fetchReason() {
     }
 
     protected String fetchDetails() {
-        // Some exception prints internal information (full class name) which is security concern
-        return emptyStringIfNull(exception.getLocalizedMessage());
+        String msg;
+        // Prevent the method from exposing internal information such as internal ip address etc. that is a security concern.
+        if (hasInternalInformation(exception)) {
+            msg = decorateMessage(exception);
+        } else {
+            msg = exception.getLocalizedMessage();
+        }
+
+        return emptyStringIfNull(msg);
     }
 
     private String emptyStringIfNull(String str) {
         return str != null ? str : "";
     }
 
+    private Boolean hasInternalInformation(Throwable t) {
+        if (t instanceof ActionTransportException) {
+            return true;
+        }
+        return false;
+    }
+
+    private String decorateMessage(Throwable t) {
+        if (t instanceof ActionTransportException) {
+            String regexIPPort = "\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}(:\\d{1,5})?";
```

Comment:
should we consider case for IPv6?

Reply:
Good point, I would have another pr to address ipv6.

Reply:
hi, @jngz-es  , this PR open for a long time , how about just add ipv6 support in this PR?

Reply:
Added ipv6 support.